### DISCUSSION TITLE
Type error when building for STM32/FreeRTOS

### DISCUSSION THREAD

**Discussion by robotoshi** [nasa/fprime#2272] [Q&A] [answered]
Hello, community!

I am a member of CalgaryToSpace, a Canadian university CubeSat team building our first satellite, and we've decided to use F\`.
We would like to use FreeRTOS on an STM32L4R5ZI and I have been working to make this port work.
I've been referencing rawinza555's community repo, [fprime-freertos](https://github.com/fprime-community/fprime-freertos/tree/v1.0) and while the build cache generation succeeds, there are a few build errors I'm running into.  
Since the repo is old, I've had to make a couple modifications to make it work, but all fairly minor.
I have an fprime (v3.3.2) project created with fprime-tools (v3.3.3) with fprime and fprime-freertos as submodules.  `fprime-freertos` is imported as a library in `settings.ini`.
If requested, I will publish my repo to show the entire setup, but for now I just want to share this compiler error, since it appears like a bug in the framework itself.

Anyway, here is the error:

```
/home/robotoshi/cubesat/CTSSAT1/fprime/Fw/Tlm/TlmPacket.cpp: In member function 'virtual Fw::SerializeStatus Fw::TlmPacket::serialize(Fw::SerializeBufferBase&) const':
/home/robotoshi/cubesat/CTSSAT1/fprime/Fw/Tlm/TlmPacket.cpp:134:67: error: call of overloaded 'serialize(const NATIVE_UINT_TYPE&)' is ambiguous
  134 |         SerializeStatus stat = buffer.serialize(this->m_numEntries);
      |                                                                   ^
In file included from /home/toshi/cubesat/CTSSAT1/fprime/Fw/Tlm/TlmPacket.cpp:7:
/home/robotoshi/cubesat/CTSSAT1/fprime/Fw/Types/Serializable.hpp:50:29: note: candidate: 'Fw::SerializeStatus Fw::SerializeBufferBase::serialize(U8)'
   50 |             SerializeStatus serialize(U8 val); //!< serialize 8-bit unsigned int
      |                             ^~~~~~~~~
...
...
...
```
...and it proceeds to list all the other overloads.
A similar message is also shown regarding the `Fw::SerializeBufferBase::deserialize()` method.

Now, I can suppress these errors by just casting the argument `NATIVE_UINT_TYPE` arg to a `U32`, but then the same error just appears in auto-generated code, which I can't easily change.

Is some compiler flag I'm missing? (I'm using the ARM toolchain included in `fprime-freetos`)
Again, just wondering if the answer is clear to those who know Fprime, or if I need to include my code for my context.

Thanks!

**Comment by thomas-bc** [comment, marked as answer]
Hi @robotoshi - glad to have you part of the community!

I believe the issue you’re seeing may be related to this one: https://github.com/nasa/fprime/issues/2036
Does that help?

**Reply by robotoshi** [reply]
Oh no, I'm sorry I totally missed that.  Thank you, I'll try it out.

**Reply by robotoshi** [reply]
Yes, that was it!  Thanks so much, I'm a little embarrassed I didn't find that issue before.
